src/cli/admin.rs

```diff
@@ -0,0 +1,219 @@
+        match self {
+            Self::Proxy(config) => config,
+            _ => panic!("attempted to unwrap proxy config when not in proxy mode"),
+        }
+    }
+
+    #[track_caller]
+    pub fn unwrap_relay(&self) -> &relay::RuntimeConfig {
+        match self {
+            Self::Relay(config) => config,
+            _ => panic!("attempted to unwrap relay config when not in relay mode"),
+        }
+    }
+
+    #[track_caller]
+    pub fn unwrap_manage(&self) -> &manage::RuntimeConfig {
+        match self {
+            Self::Manage(config) => config,
+            _ => panic!("attempted to unwrap relay config when not in relay mode"),
+        }
+    }
+
+    pub fn idle_request_interval_secs(&self) -> u64 {
+        match self {
+            Self::Proxy(config) => config.idle_request_interval_secs,
+            Self::Agent(config) => config.idle_request_interval_secs,
+            Self::Relay(config) => config.idle_request_interval_secs,
+            _ => IDLE_REQUEST_INTERVAL_SECS,
+        }
+    }
+
+    pub fn server(
+        &self,
+        config: Arc<Config>,
+        address: Option<std::net::SocketAddr>,
+    ) -> tokio::task::JoinHandle<Result<(), hyper::Error>> {
+        let address = address.unwrap_or_else(|| (std::net::Ipv6Addr::UNSPECIFIED, PORT).into());
+        let health = Health::new();
+        tracing::info!(address = %address, "Starting admin endpoint");
+
+        let mode = self.clone();
+        let make_svc = make_service_fn(move |_conn| {
+            let config = config.clone();
+            let health = health.clone();
+            let mode = mode.clone();
+            async move {
+                let config = config.clone();
+                let health = health.clone();
+                let mode = mode.clone();
+                Ok::<_, Infallible>(service_fn(move |req| {
+                    let config = config.clone();
+                    let health = health.clone();
+                    let mode = mode.clone();
+                    async move { Ok::<_, Infallible>(mode.handle_request(req, config, health)) }
+                }))
+            }
+        });
+
+        tokio::spawn(HyperServer::bind(&address).serve(make_svc))
+    }
+
+    fn is_healthy(&self, config: &Config) -> bool {
+        match &self {
+            Self::Proxy(proxy) => proxy.is_healthy(config),
+            Self::Agent(agent) => agent.is_healthy(),
+            Self::Manage(manage) => manage.is_healthy(),
+            Self::Relay(relay) => relay.is_healthy(),
+        }
+    }
+
+    fn handle_request(
+        &self,
+        request: Request<Body>,
+        config: Arc<Config>,
+        health: Health,
+    ) -> Response<Body> {
+        match (request.method(), request.uri().path()) {
+            (&Method::GET, "/metrics") => collect_metrics(),
+            (&Method::GET, "/live" | "/livez") => health.check_healthy(),
+            (&Method::GET, "/ready" | "/readyz") => check_readiness(|| self.is_healthy(&config)),
```

Comment:
> rather than making/assuming they are the same thing - since I would recommend we default to readiness being false, but liveness being true.

Maybe the code isn't clear, but they aren't the same thing. The `health.check_healthy()` check only operates on whether the program has panicked. Where as `Admin::is_healthy` is service specific readiness.

Reply:
Ooooh! You are right, I totally saw "healthy" and read that as _liveness_, since the `health.check_healthy()` has the same name.

In which case, let's rename things to they align properly, so something like:

`health.check_healthy(),` ➡️  `health.check_liveness()` 
`RuntimeConfig::is_healthy(...)` ➡️  `RuntimeConfig::is_ready(...)`

So then both use the right words for everything. Sound good?